### DISCUSSION TITLE
Search analytics self-hosted

### DISCUSSION THREAD

**Discussion by AndrewMarques** [meilisearch/product#767] [Cloud Feedback]
Hi,

Is there any way to view search analytics using the self-hosted project? I would like to know the most searched terms, number of searches, etc.

**Comment by macraig** [comment]
Hello @AndrewMarques, currently search analytics is only available on our Cloud offering. This is because the feature is built using a third-party service that is outside of the open-source engine.

**Comment by mrancilio** [comment]
I have created a Python script that reads the query logs from Meilisearch, sends them via API to OpenAI which, using text-embedding-3-large (small is not suitable for Italian), creates a JSON file with clusters of the most used keywords. Within the clusters, I can see the individual searches. This is then displayed in a table in Grafana, ordered by the most used keywords.

Here is an example of the JSON:


"keywords": [
        {
            "keyword": "sardegna vermentini",
            "count": 128,
            "variations": [
                "sardegna vermentini",
                "sardegna vermentino",
                "sardegna vermentino blu",
                "vermentino",
                "vermentino blu",
                "vermentino santa",
                "vermentino santa maria",
                "vermentino santa maria palma",
                "blu vermentino",
                "vermentino kar",
                "vermentino karmis",
                "vermentino gall",
                "vermentino gallura",
                "vermentino sardegna",
                "vermentino sarde",
                "vermentino antinori",
                "vermentino sar",
                "vermentino argiolas",
                "vermentino pesce",
                "vermentino biolo",
                "vermentino biologico",
                "vermentino rosso",
                "vermentino sup",
                "vermentino super",
                "vermentino superiore",
                "vermentino sardo",
                "vermentino capi",
                "vermentino capi hera",
                "vermentino capi her",
                "vermentino cap"
            ]
        },
        {
            "keyword": "gin",
            "count": 124,
            "variations": [
                "gin"
            ]
        },
